pkg/apiserver/registry/stats/multicastgroup/rest.go

```diff
@@ -0,0 +1,150 @@
+	return &statsv1alpha1.MulticastGroup{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &statsv1alpha1.MulticastGroupList{}
+}
+
+func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.NetworkPolicyStats) {
+		return &statsv1alpha1.MulticastGroupList{}, nil
+	}
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroupList{}, nil
+	}
+	multicastGroups := &statsv1alpha1.MulticastGroupList{
+		Items: r.statsProvider.ListMulticastGroups(),
+	}
+	return multicastGroups, nil
+}
+
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.NetworkPolicyStats) {
+		return &statsv1alpha1.MulticastGroup{}, nil
+	}
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroup{}, nil
+	}
+	multicastGroup, exists := r.statsProvider.GetMulticastGroup(name)
+	if !exists {
+		return nil, errors.NewNotFound(statsv1alpha1.Resource("multicastgroup"), name)
+	}
+	return multicastGroup, nil
+}
+
+func (r *REST) ConvertToTable(ctx context.Context, obj runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
+	table := &metav1.Table{
+		ColumnDefinitions: []metav1.TableColumnDefinition{
+			{Name: "Group", Type: "string", Format: "name", Description: "IP of multicast group."},
+			{Name: "Pods", Type: "string", Description: "List of Pods the has joined the multicast group."},
+		},
+	}
+	if m, err := meta.ListAccessor(obj); err == nil {
+		table.ResourceVersion = m.GetResourceVersion()
+		table.Continue = m.GetContinue()
+		table.RemainingItemCount = m.GetRemainingItemCount()
+	} else {
+		if m, err := meta.CommonAccessor(obj); err == nil {
+			table.ResourceVersion = m.GetResourceVersion()
+		}
+	}
+
+	var err error
+	table.Rows, err = metatable.MetaToTableRow(obj, func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+		stats := obj.(*statsv1alpha1.MulticastGroup)
+		return []interface{}{stats.Group, formatPodReferenceList(stats.Pods)}, nil
+	})
+	return table, err
+}
+
+// formatPodReferenceList formats a list of PodReference and cut it if it encodes more than 3 pods.
+// Example: formatPodReferenceList(pods with len 13)
+// apodNamespace/apodName,bpodNamespace/bpodName,bpodNamespace/bpodName + 10 more...
+// The implementation of this func is similar to
+// https://github.com/kubernetes/kubernetes/blob/bb7b0b6a91e9f365025d496cb267817e777c872b/pkg/printers/internalversion/printers.go#L667.
+func formatPodReferenceList(pods []statsv1alpha1.PodReference) string {
+	list := []string{}
+	max := 3
+	more := false
+	count := 0
+	for _, endpoint := range pods {
+		if len(list) < max {
+			list = append(list, k8s.NamespacedName(endpoint.Namespace, endpoint.Name))
+		} else if len(list) == max {
+			more = true
+		}
+		count++
```

Comment:
The function you refer to has nested loops so it needs to calculate `list, more, count` in that way, but our case is simpler so no point to copy the code exactly and sacrifice performance

Reply:
Could you highlight the nested loop you are referring to? I don't think it is nested.

Reply:
I said "the function you refer to" has nested loops: https://github.com/kubernetes/kubernetes/blob/bb7b0b6a91e9f365025d496cb267817e777c872b/pkg/printers/internalversion/printers.go#L672-L681, not your own code.

Since you don't have nested loop, do you really need to traverse all items to get `count`, and need `if len(list) == max` to get `more`?
```
count := len(pods)
if count > max {
    pods = pods[:max]
}
list := make([]string{}, 0, len(pods))
for _, pod := range pods {
    list = append(list, k8s.NamespacedName(pod.Namespace, pod.Name))
}
ret := strings.Join(list, ",")
if count > max {
    return fmt.Sprintf("%s + %d more...", ret, count-max)
}
if ret == "" {
    ret = "<none>"
}
return ret
```

Reply:
updated, thanks. The code you provide is more concise.